prediction_market_agent/agents/crewai_subsequential_agent/crewai_agent_subquestions.py

```diff
@@ -0,0 +1,192 @@
+            agent=self.researcher,
+            expected_output=RESEARCH_OUTCOME_OUTPUT,
+        )
+        task_create_probability_for_one_outcome = Task(
+            description=PROBABILITY_FOR_ONE_OUTCOME_PROMPT,
+            expected_output=PROBABILITY_CLASS_OUTPUT,
+            agent=self.predictor,
+            output_json=ProbabilityOutput,
+            context=[task_research_one_outcome],
+        )
+        crew = Crew(
+            agents=[self.researcher, self.predictor],
+            tasks=[task_research_one_outcome, task_create_probability_for_one_outcome],
+            verbose=2,
+            process=Process.sequential,
+        )
+
+        result = crew.kickoff(inputs={"sentence": sentence})
+        return ProbabilityOutput.model_validate_json(result)
+
+    def generate_final_decision(
+        self, outcomes_with_probabilities: list[t.Tuple[str, ProbabilityOutput]]
+    ) -> ProbabilityOutput:
+        task_final_decision = Task(
+            description=FINAL_DECISION_PROMPT,
+            agent=self.predictor,
+            expected_output=PROBABILITY_CLASS_OUTPUT,
+            output_json=ProbabilityOutput,
+        )
+
+        crew = Crew(
+            agents=[self.predictor],
+            tasks=[task_final_decision],
+            verbose=2,
+        )
+
+        crew.kickoff(
+            inputs={
+                "outcomes_with_probabilities": [
+                    (i[0], i[1].dict()) for i in outcomes_with_probabilities
+                ],
+                "number_of_outcomes": len(outcomes_with_probabilities),
+                "outcome_to_assess": outcomes_with_probabilities[0][0],
+            }
+        )
+        return ProbabilityOutput.model_validate_json(
+            task_final_decision.output.raw_output
+        )
+
+    def answer_binary_market(self, question: str) -> ProbabilityOutput:
+        outcomes = self.split_research_into_outcomes(question)
+        print("outcomes ", outcomes)
+
+        outcomes_with_probs = []
+        task_map = {}
+        for outcome in outcomes.outcomes:
+            tasks_for_outcome = self.build_tasks_for_outcome(
+                input_dict={"sentence": outcome}
+            )
+            task_map[outcome] = tasks_for_outcome
+
+        # flatten nested list
+        all_tasks = sum(task_map.values(), [])
+        crew = Crew(
+            agents=[self.researcher, self.predictor],
+            tasks=all_tasks,
+            verbose=2,
+            process=Process.sequential,
+        )
+
+        # crew.kickoff doesn't finish all async tasks when done.
+        crew.kickoff()
+
+        # We parse individual task results to build outcomes_with_probs
+        for outcome, tasks in task_map.items():
+            try:
+                prediction_result = ProbabilityOutput.model_validate_json(
+                    tasks[1].output.raw_output
+                )
+            except Exception as e:
+                print("Could not parse result as ProbabilityOutput ", e)
```

Comment:
```suggestion
                logger.error(f"Could not parse result as ProbabilityOutput {e} ")
```

Otherwise, alerts won't notify us about errors.